### DISCUSSION TITLE
Library Manager

### DISCUSSION THREAD

**Discussion by Paul8043** [bernhard-42/vscode-ocp-cad-viewer#24] [Q&A]
I appreciate the OCP-entry in the activity-bar of VS-Code. The Viewer-Manager and Library-Manager make some things much more easier. I have tried to do all installation-work with the Library-Manager only and that worked fine.

Now my question:

For ocp_vscode 1.2.2 I got a windows-notification (message in the lower-right corner) which told me that a new version is available. By clicking inside the message-box the installation could be started. I have rejected this offer, because it is not so clear what windows will do! But if I would have accepted the offer, leads this to the same result as green-down-arrow of the ocp-vscode-entry  of the Library-Manager? How can I be sure that windows selects the right modality (none, pip, conda, mamba)? What is your recommendation?

Is there an easy way to tell the Library-Manager: Update all Libraries with the previously used modalities?

**Comment by bernhard-42** [comment]
> But if I would have accepted the offer, leads this to the same result as green-down-arrow of the ocp-vscode-entry of the Library-Manager

yes

> Is there an easy way to tell the Library-Manager: Update all Libraries with the previously used modalities?

ocp_vscode will automatically update ocp-tessellate. Then there is just build123d remaining.
I don't want to over-engineer this installation part. 


btw. you can create your own install command in the VSCode OCP CAD Viewer Setting dialog:

```json
    "OcpCadViewer.advanced.installCommands": {
        "matplotlib": {
            "pip": [
                "{python} -m pip install matplotlib"
            ]
        },

...
```
Remember to press refresh in the library manager

Then: 
![image](https://github.com/bernhard-42/vscode-ocp-cad-viewer/assets/954408/9e35ee5c-5227-4de9-87fb-6a6ffbcfcb17)

**Reply by Paul8043** [reply]
> Remember to press refresh in the library manager

Sorry, I have overseen this button.

**Reply by Paul8043** [reply]
Now I remember. I had skipped this button, because it does not generate any user-feedback. How can I see what the button tries to do and whether the action was successful or not?
